### DISCUSSION TITLE
Is there a way to print out what the routes look like?

### DISCUSSION THREAD

**Discussion by gnapse** [kiliman/remix-flat-routes#120] [Q&A] [answered]
I'm looking for a command similar to what exists in rails with the `rails routes` command (see [Listing existing routes](https://guides.rubyonrails.org/routing.html#listing-existing-routes) in the Rails documentation).

I'm thinking of something that one could then set up to run via a `npm run routes` command.

**Comment by kiliman** [comment, marked as answer]
Yes. From the CLI:

```shell
npx remix routes
```
You can add the `--json` flag as well.

**Reply by gnapse** [reply]
Wow, thanks. Now I feel dumb for not having found this documented somewhere. I'm sure it is. Thanks again!

**Reply by kiliman** [reply]
I think it used to be, but I can't find it anywhere in the documenation. I think when they split out the CLI for Remix Classic compiler and the new Vite compiler, those commands may have been accidentally cut.
